File: bindings/lua/Polycode/ShaderPass.lua
class "ShaderPass"


function ShaderPass:__getvar(name)
	if name == "wireframe" then
		return Polycode.ShaderPass_get_wireframe(self.__ptr)
	elseif name == "blendingMode" then
		return Polycode.ShaderPass_get_blendingMode(self.__ptr)
	end
end

function ShaderPass:__setvar(name,value)
	if name == "wireframe" then
		Polycode.ShaderPass_set_wireframe(self.__ptr, value)
		return true
	elseif name == "blendingMode" then
		Polycode.ShaderPass_set_blendingMode(self.__ptr, value)
		return true
	end
	return false
end
function ShaderPass:ShaderPass(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.ShaderPass(unpack(arg))
	end
end

function ShaderPass:getShaderBinding()
	local retVal =  Polycode.ShaderPass_getShaderBinding(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<ShaderBinding>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function ShaderPass:__delete()
	if self then Polycode.delete_ShaderPass(self.__ptr) end
end

File: bindings/javascript/Polycode/ShaderPass.js
function ShaderPass() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.ShaderPass()
	}
	Object.defineProperties(this, {
		'wireframe': { enumerable: true, configurable: true, get: ShaderPass.prototype.__get_wireframe, set: ShaderPass.prototype.__set_wireframe},
		'blendingMode': { enumerable: true, configurable: true, get: ShaderPass.prototype.__get_blendingMode, set: ShaderPass.prototype.__set_blendingMode}
	})
}


ShaderPass.prototype.__get_wireframe = function() {
	return Polycode.ShaderPass__get_wireframe(this.__ptr)
}

ShaderPass.prototype.__set_wireframe = function(val) {
	Polycode.ShaderPass__set_wireframe(this.__ptr, val)
}

ShaderPass.prototype.__get_blendingMode = function() {
	return Polycode.ShaderPass__get_blendingMode(this.__ptr)
}

ShaderPass.prototype.__set_blendingMode = function(val) {
	Polycode.ShaderPass__set_blendingMode(this.__ptr, val)
}

Duktape.fin(ShaderPass.prototype, function (x) {
	if (x === ShaderPass.prototype) {
		return;
	}
	Polycode.ShaderPass__delete(x.__ptr)
})

ShaderPass.prototype.getShaderBinding = function() {
	var retVal = new ShaderBinding("__skip_ptr__")
	retVal.__ptr = Polycode.ShaderPass_getShaderBinding(this.__ptr)
	return retVal
}
